CrustTests/CRMapperTests.swift

```diff
@@ -21,7 +21,8 @@ class MockMap: Mapping, Adapter {
         catchMapping!(toMap, context)
     }
     
-    func mappingBegins() throws { }
+    var mappingDidBegin: Bool = false
+    func mappingBegins() throws { self.mappingDidBegin = true }
```

Comment:
Ya, probably good to change that now. Though I'm wondering if it should just be more explicit too. What do you think about `mappingWillBegin` and `mappingDidEnd`. Also, I'm kinda struggling with `mappingDidBegin` as a property (and with those two changes it would be even more confusing named this way). Maybe `isMapping`? Or maybe something even more concrete like `isInTransaction` or something - though I feel like that's maybe too specific.